h/presenters/annotation_json.py

```diff
@@ -60,10 +58,14 @@ def _get_read_permission(self):
             # It's not shared so only the owner can read it
             return self.annotation.userid
 
-        if security.Everyone in principals_allowed_by_permission(
-            AnnotationContext(self.annotation), Permission.Annotation.READ
+        # If we can access this without an identity, anyone in the world
+        # can read this. We optimise by checking if we were going to write
+        # "group:__world__" anyway, to save the permission check.
+        if self.annotation.groupid == "__world__" or identity_permits(
+            identity=None,
+            context=AnnotationContext(self.annotation),
+            permission=Permission.Annotation.READ,
         ):
```

Comment:
Here we need to ask a theoretical question of the permissions system. "Would an identity of None be allowed to do this?"

This isn't the same as asking whether _we_ could do it (unless our identity happens to be None). This is why we don't go through the policy.